common/threadpool.cc

```diff
@@ -45,6 +45,11 @@ void Threadpool::Shutdown() {
   for (auto& worker : workers_) {
     if (worker.joinable()) worker.join();
   }
+
+  // Discard all completed tasks.
+  absl::MutexLock lock(&completed_tasks_mutex_);
+  std::queue<std::unique_ptr<Task>> empty;
+  std::swap(completed_tasks_, empty);
```

Comment:
Why the swap? To me this is implying that more work needs to be done, while move assignment from a temporary default constructed queue should be enough:

 `completed_tasks_ = {};`

Or is this missing some return statement?

Reply:
Swapping deletes memory, whereas the move assignment doesn't. Not really super relevant, since Shutdown() is right now always called when the pool is deleted, anyway.

```
  std::vector<int> vec;
  vec.resize(100, 123);
  printf("%zu\n", vec.capacity());
  vec = {};
  printf("%zu\n", vec.capacity());
  std::vector<int> empty;
  std::swap(vec, empty);
  printf("%zu\n", vec.capacity());
```
outputs
```
100
100
0
```